nasa/harmony
# HARMONY-1442: Update AWS SDK from V2 to V3

## Jira Issue ID
HARMONY-1442

## Description
Updates the AWS SDK from V2 to V3

## Local Test Steps
This one needs a lot of regression testing. Everything that potentially uses AWS (S3, ECR, STS, SQS).

1. Remove the `/built` directory from all three of the tasks (query-cmr, service-runner, giovanni-adapter).
2. nvm use && npm install
3. Rebuild all of the task images
4. In each of the 3 tasks subdirectories: `npm run build`
5. Restart your k8s services: `bin/restart-services`
6. Start up harmony and test

Test out that a request completes successfully.
In the workflow-ui make sure you can get to the logs (not the metrics button - just the in browser logs)
Make sure the /versions endpoint shows services that are using ECR and shows timestamps (sandbox only).
Make sure the /cloud-access and /cloud-access.sh endpoints return credentials successfully.
Turn on queues and make sure they work as expected and requests complete successfully.
Make sure a request that uses a shapefile works and correctly uses the shapefile.

Try to think of other things I'm not thinking of. You can also deploy to sandbox to test with real AWS rather than just localstack.

## PR Acceptance Checklist
